core/src/main/java/org/apache/gravitino/catalog/SchemaOperationDispatcher.java

```diff
@@ -201,68 +213,73 @@ public Schema loadSchema(NameIdentifier ident) throws NoSuchSchemaException {
   @Override
   public Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
       throws NoSuchSchemaException {
-    validateAlterProperties(ident, HasPropertyMetadata::schemaPropertiesMetadata, changes);
 
     NameIdentifier catalogIdent = getCatalogIdentifier(ident);
-    Schema alteredSchema =
-        doWithCatalog(
-            catalogIdent,
-            c -> c.doWithSchemaOps(s -> s.alterSchema(ident, changes)),
-            NoSuchSchemaException.class);
-
-    // If the Schema is maintained by the Gravitino's store, we don't have to alter again.
-    boolean isManagedSchema = isManagedEntity(catalogIdent, Capability.Scope.SCHEMA);
-    if (isManagedSchema) {
-      return EntityCombinedSchema.of(alteredSchema)
-          .withHiddenProperties(
-              getHiddenPropertyNames(
+    return TreeLockUtils.doWithTreeLock(
+        NameIdentifier.of(ident.namespace().levels()),
```

Comment:
I'm curious why do we need to add the write in the catalog level? I think it should be fine to add the write lock in the schema level, am I right?

Reply:
The following scenario will cause problems if we use write lock in the schema level，assuming the schema name is `schema1`


TheadA                                                      ThreadB

hold schema1 write lock

rename schema1 to schema2                

store schema2 to storage                    hold schema2 write lock   

block due to some reason                   rename schema2 to schema1
                                                               
 return                                                    store schema2 to storage 
                                                               return

Reply:
This problem only happens for rename, I think we don't support schema rename. Besides, we can treat rename and other change separately, if the changes contain `rename`, then we use parent level write lock, otherwise we use its own write lock. WDYT?

Reply:
Let me verify this and check if this can be applied to catalogs, tables, and other objects.

Reply:
I changed the code and determined if we need to lock the parent or itself.